trin-core/src/portalnet/find/iterators/findnodes.rs

```diff
@@ -232,11 +196,13 @@ where
                         return QueryState::WaitingAtCapacity;
                     }
                 }
-
                 QueryPeerState::Waiting(timeout) => {
                     if now >= *timeout {
                         // Peers that don't respond within timeout are set to `Failed`.
-                        debug_assert!(self.num_waiting > 0);
+                        assert!(
```

Comment:
Also noticed the usage of `assert!` in `FindContentQuery` - why the decision to panic if a query is unsuccessful / buggy instead of something like a failed query state that's handled?

Reply:
This logic was carried over from `sigp/discv5`, albeit with `debug_assert`, which I think we can change back to. My understanding is that this type of error could represent a non-recoverable error, because if it can happen for one query, it can happen for many queries.

If we did not have the assertion, and we had `num_waiting == 0`, then we would panic on 'attempt to subtract with overflow'. So it seems that this assertion is a more useful panic message in the case that we run into such a situation.

I think we can assert that this is something that should never happen, because it represents a bug in our own logic. What do you think? If we keep the assertion, should go back to `debug_assert`? @njgheorghita